### PR TITLE
updated sfx to otel for py instrumentation

### PR DESCRIPTION
**Pull request by harnitsignalfx**

### CODE REVIEW

File: src/emailservice/email_server.py [hunk 1]
```diff
@@ -28,39 +28,13 @@
...
-from opentelemetry import trace
-from opentelemetry import propagators
-from opentelemetry.sdk.trace import TracerProvider
-from opentelemetry.exporter import zipkin
-from opentelemetry.sdk.trace.export import BatchExportSpanProcessor
-from opentelemetry.sdk.trace.propagation.b3_format import B3Format
-from opentelemetry.instrumentation.grpc import GrpcInstrumentorServer
-from opentelemetry.instrumentation.grpc import GrpcInstrumentorClient
-from opentelemetry.instrumentation.grpc.grpcext import intercept_server
+from opentelemetry.instrumentation.grpc import server_interceptor
```
Comment (owais) [inline src/emailservice/email_server.py:31]: I don't think you should need this as `start_tracing()` will auto instrument the service.

File: src/emailservice/email_server.py [hunk 2]
```diff
@@ -94,7 +68,7 @@ def Watch(self, request, context, send_response_callback=None):
 
 def start():
   server = grpc.server(futures.ThreadPoolExecutor(max_workers=10),
-                       interceptors=tuple())
+                        interceptors=[server_interceptor()])
```
Comment (owais) [inline src/emailservice/email_server.py:71]: I'd try without this first and see if the service gets auto-instrumented.

Comment (harnitsignalfx) [reply src/emailservice/email_server.py:71]: ack, I did try it initially and it didn't work. But worth revisiting as I might've missed something.

File: src/emailservice/requirements.txt [hunk 3]
```diff
@@ -11,14 +11,11 @@ grpcio==1.30.0            # via -r requirements.in, grpcio-health-checking, open
...
-opentelemetry-api==0.12b0  # via -r requirements.in, opentelemetry-exporter-zipkin, opentelemetry-instrumentation, opentelemetry-instrumentation-dbapi, opentelemetry-instrumentation-grpc, opentelemetry-instrumentation-requests, opentelemetry-instrumentation-sqlite3, opentelemetry-sdk
-opentelemetry-exporter-zipkin==0.12b0  # via -r requirements.in
-opentelemetry-instrumentation-dbapi==0.12b0  # via opentelemetry-instrumentation-sqlite3
-opentelemetry-instrumentation-grpc==0.12b0  # via -r requirements.in
-opentelemetry-instrumentation-requests==0.12b0  # via -r requirements.in
-opentelemetry-instrumentation-sqlite3==0.12b0  # via -r requirements.in
-opentelemetry-instrumentation==0.12b0  # via -r requirements.in, opentelemetry-instrumentation-dbapi, opentelemetry-instrumentation-requests, opentelemetry-instrumentation-sqlite3
-opentelemetry-sdk==0.12b0  # via -r requirements.in, opentelemetry-exporter-zipkin, opentelemetry-instrumentation-grpc
+opentelemetry-instrumentation-dbapi==0.19b0  # via opentelemetry-instrumentation-sqlite3
+opentelemetry-instrumentation-grpc==0.19b0  # via -r requirements.in
```
[the remark this answers is not in the document]
Comment (harnitsignalfx) [reply src/emailservice/requirements.txt:15]: updated! to what I think the values should be :)

File: src/emailservice/email_server.py (hunk 2 above)
Comment (harnitsignalfx) [reply src/emailservice/email_server.py:71]: thank you! it does work ! removing the interceptors.

File: src/emailservice/email_server.py (hunk 1 above)
Comment (harnitsignalfx) [reply src/emailservice/email_server.py]: ack, tested and updated! ty!

File: src/emailservice/requirements.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (owais) [reply src/emailservice/requirements.txt:15]: requirements.in is what you should edit. requirements.txt is auto generated from it. I don't remember how :)

**Review by owais — APPROVED** [review]

**Pull request merged**